yarn-project/circuit-types/src/p2p/mocks.ts

```diff
@@ -0,0 +1,38 @@
+import { makeHeader } from '@aztec/circuits.js/testing';
+import { Fr } from '@aztec/foundation/fields';
+
+import { type PrivateKeyAccount } from 'viem';
+import { generatePrivateKey, privateKeyToAccount } from 'viem/accounts';
+
+import { TxHash } from '../tx/tx_hash.js';
+import { BlockAttestation } from './block_attestation.js';
+import { BlockProposal } from './block_proposal.js';
+import { Signature } from './signature.js';
+
+export const makeBlockProposal = async (signer?: PrivateKeyAccount): Promise<BlockProposal> => {
+  signer = signer || randomSigner();
+
+  const blockHeader = makeHeader(1);
+  const archive = Fr.random();
+  const txs = [0, 1, 2, 3, 4, 5].map(() => TxHash.random());
+  const digest = `0x${archive.toBuffer().toString('hex')}` as `0x${string}`;
+  const signature = Signature.from0xString(await signer.signMessage({ message: { raw: digest } }));
+
+  return new BlockProposal(blockHeader, archive, txs, signature);
+};
+
+export const makeBlockAttestation = async (signer?: PrivateKeyAccount): Promise<BlockAttestation> => {
```

Comment:
It feels weird here that the attestation might be on random things, at least if we wanted to create multiple attestations that we gotta check are same block etc 🤷

Reply:
https://github.com/AztecProtocol/aztec-packages/issues/8028